nd4j/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/api/ops/custom/FakeQuantWithMinMaxVarsPerChannel.java

```diff
@@ -21,30 +21,60 @@
 import org.nd4j.linalg.api.buffer.DataType;
 import org.nd4j.linalg.api.ndarray.INDArray;
 import org.nd4j.linalg.api.ops.DynamicCustomOp;
+import org.tensorflow.framework.AttrValue;
+import org.tensorflow.framework.GraphDef;
+import org.tensorflow.framework.NodeDef;
 
 import java.util.Collections;
 import java.util.List;
+import java.util.Map;
 
 public class FakeQuantWithMinMaxVarsPerChannel extends DynamicCustomOp {
+    protected boolean narrowRange;
+    protected int numBits;
+
     public FakeQuantWithMinMaxVarsPerChannel() {}
 
-    public FakeQuantWithMinMaxVarsPerChannel(INDArray x, INDArray min, INDArray max) {
+    public FakeQuantWithMinMaxVarsPerChannel(INDArray x, INDArray min, INDArray max, int num_bits, boolean narrow) {
         Preconditions.checkArgument(min.isVector() && max.isVector() &&
                         min.length() == max.length(),
                 "FakeQuantWithMinMaxVarsPerChannel: min and max should be 1D tensors with the same length");
-        inputArguments.add(x);
-        inputArguments.add(min);
-        inputArguments.add(max);
+        addInputArgument(x,min,max);
+        addIArgument(num_bits);
+        addBArgument(narrow);
     }
 
-    public FakeQuantWithMinMaxVarsPerChannel(INDArray x, INDArray min, INDArray max,
-                                             INDArray output) {
-        this(x,min,max);
-        outputArguments.add(output);
+    public FakeQuantWithMinMaxVarsPerChannel(INDArray x, INDArray min, INDArray max, int num_bits) {
+        Preconditions.checkArgument(min.isVector() && max.isVector() &&
+                        min.length() == max.length(),
+                "FakeQuantWithMinMaxVarsPerChannel: min and max should be 1D tensors with the same length");
+        addInputArgument(x,min,max);
+        addIArgument(num_bits);
+        addBArgument(false);
     }
 
-    public FakeQuantWithMinMaxVarsPerChannel(SameDiff sameDiff, SDVariable x, SDVariable min, SDVariable max) {
+    public FakeQuantWithMinMaxVarsPerChannel(INDArray x, INDArray min, INDArray max, boolean narrow) {
+        Preconditions.checkArgument(min.isVector() && max.isVector() &&
+                        min.length() == max.length(),
+                "FakeQuantWithMinMaxVarsPerChannel: min and max should be 1D tensors with the same length");
+        addInputArgument(x,min,max);
+        addIArgument(8);
+        addBArgument(narrow);
+    }
+
+    public FakeQuantWithMinMaxVarsPerChannel(INDArray x, INDArray min, INDArray max) {
+        Preconditions.checkArgument(min.isVector() && max.isVector() && min.length() == max.length(),
+                "FakeQuantWithMinMaxVarsPerChannel: min and max should be 1D tensors with the same length");
+        addInputArgument(x,min,max);
+        addIArgument(8);
+        addBArgument(false);
+    }
```

Comment:
These 3 constructors could be cleaned up into:
```
public FakeQuantWithMinMaxVarsPerChannel(INDArray x, INDArray min, INDArray max, int num_bits){
    this(x, min, max, num_bits, false);
}

public FakeQuantWithMinMaxVarsPerChannel(INDArray x, INDArray min, INDArray max, boolean narrow) {
    this(x, min, max, 8, narrow);
}

public FakeQuantWithMinMaxVarsPerChannel(INDArray x, INDArray min, INDArray max, int num_bits, boolean narrow) {
    ...
}
```

Reply:
Fixed.